themes/bootstrap3/js/facets.js

```diff
@@ -116,18 +116,276 @@ VuFind.register('facetList', function FacetList() {
+          // If the parameter is already present we update it
+          updateInitialParams(input.name, input.value);
+        } else {
+          addedRangeParams.push(encodeURI(input.name + '=' + input.value));
+        }
+        if (input.value !== '') {
+          allEmptyRangeParams = false;
+        }
+      }
+
+      // If at least one parameter is not null we continue the routine for the final URL
+      if (allEmptyRangeParams === false) {
+        globalAddedParams = globalAddedParams.concat(addedRangeParams);
+        inputs = form.querySelectorAll('input');
+        for (let input of inputs) {
+          if (rangeSelectorId.startsWith(input.value)) {
+            let dateRangeParam = encodeURI(input.name + '=' + input.value);
+            if (!window.location.search.match(dateRangeParam)) {
+              globalAddedParams.push(dateRangeParam);
+            }
+          }
+        }
+      }
+    }
+  }
+
+  function getHrefWithNewParams() {
+    handleRangeSelector();
+    // Unique parameters
+    initialRawParams = initialRawParams.filter(function onlyUnique(value, index, array) {
+      return array.indexOf(value) === index;
+    });
+    // Removing parameters
+    initialRawParams = initialRawParams.filter(function tmp(obj) { return !globalRemovedParams.includes(obj); });
+    // Adding parameters
+    initialRawParams = initialRawParams.concat(globalAddedParams);
+    return window.location.pathname + '?' + initialRawParams.join('&');
+  }
+
+  function applyMultiFacetsSelection() {
+    defaultContext.getElementsByClassName('applyMultiFacetsSelection')[0]
+      .removeEventListener('click', applyMultiFacetsSelection);
+    if (callbackOnApply instanceof Function) {
+      callbackOnApply();
+    }
+    window.location.assign(getHrefWithNewParams());
+  }
+
+  function rangeSelectorInit() {
+    document.querySelectorAll('div.facet form .date-fields').forEach((elem) => {
+      rangeSelectorIds.push(elem.parentElement.id);
+      elem.parentElement.addEventListener('submit', function switchAction(e) {
+        if (isMultiFacetsSelectionActivated) {
+          e.preventDefault();
+        }
+      });
+    });
+  }
+
+  function facetSelectionStyling(elem) {
+    if (elem.classList.contains('exclude')) {
+      elem.classList.toggle('selected');
+    } else {
+      if (elem.classList.contains('facet')) {
+        elem.classList.toggle('active');
+      } else if (elem.parentElement.classList.contains('facet')) {
+        elem.parentElement.classList.toggle('active');
+      }
+
+      let icon = elem.querySelector('.icon');
+      if (icon !== null) {
+        let checked = icon.attributes['data-checked'].value === 'true';
+        let attrs = [];
+        attrs['class'] = 'icon-link__icon';
+        attrs['data-checked'] = (checked ? 'false' : 'true');
+        icon.outerHTML = VuFind.icon( checked? 'facet-unchecked' : 'facet-checked', attrs);
```

Comment:
Please take a look at the Files tab of the PR -- you'll see a few Javascript style issues (such as the missing space before the `?` operator here) highlighted by the automatic checking process. Can you please fix these?